packages/caliper-core/lib/worker/workload/declarative/value-providers/formatted-string-value-provider.js

```diff
@@ -0,0 +1,108 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const ValueProviderInterface = require('./value-provider-interface');
+/**
+ * Value class provider for formatted strings based on other value providers.
+ */
+class FormattedStringValueProvider extends ValueProviderInterface {
+    /**
+     * Initialize an instance of FormattedStringValueProvider
+     * @param {object} options The user provided options for value provider
+     * @param {object} variables Store of variables managed by workload module
+     * @param {object} parameters Store of workload parameters provided by the user through the round configuration
+     * @param {object} valueProviderFactory ValueProviderFactory object reference
+     */
+    constructor(options, variables, parameters, valueProviderFactory) {
+        super(options, variables, parameters, valueProviderFactory);
+
+        if (this.options === undefined) {
+            throw new Error(`Incorrect options value: ${this.options}`);
+        }
+
+        if (this.valueProviderFactory === undefined) {
+            throw new Error(`Incorrect valueProviderFactory value: ${this.valueProviderFactory}`);
+        }
+
+        if (
+            this.options.format === undefined ||
+            typeof this.options.format !== 'string'
+        ) {
+            throw new Error(`Invalid format value: ${this.options.format}`);
+        }
+
+        if (!Array.isArray(this.options.parts)) {
+            throw new Error(`Incorrect parts value: ${this.options.parts}`);
+        }
+
+        this.format = this.options.format;
+        this.parts = this.options.parts;
+        const partLength = this.parts.length;
+
+        for (let i = 1; i <= partLength; i++) {
```

Comment:
This check is not needed, since Line#78 does the same in a nicer way. Also update the corresponding expected error messages in the tests